MdeModulePkg/Universal/Console/ConSplitterDxe/ConSplitter.c

```diff
@@ -30,6 +30,9 @@ WITHOUT WARRANTIES OR REPRESENTATIONS OF ANY KIND, EITHER EXPRESS OR IMPLIED.
 
 #include "ConSplitter.h"
 
+// MU_CHANGE: do not perform set mode during device add flow if the required mode is already set

```

Comment:
Is there a reason this isn't generalized to only *ever* set the mode if it doesn't already match?

Reply:
This function gets called in 2 cases - internal call when adding a new device (which flow I'm modifying) and protocol call that lands here when the user of the protocol explicitly wants to SetMode. According to UEFI spec, SetMode is supposed to clear the screen if the mode is already set, so we're keeping that behavior. However, the spec doesn't specify the ConSplitter behavior on addition of new ConOut devices - and I can't think of a reason we'd want to clear the screens on all consoles when adding a new ConOut device to the ConSplitter.